spatialmath/geometry_utils.go

```diff
@@ -81,6 +81,24 @@ func PlaneNormal(p0, p1, p2 r3.Vector) r3.Vector {
 	return p1.Sub(p0).Cross(p2.Sub(p0)).Normalize()
 }
 
+// Bounding sphere returns a spherical geometry centered on the point (0, 0, 0) that will encompass the given geometry
+// if it were to be rotated 360 degrees about the Z axis.  The label of the new geometry is inherited from the given one.
+func BoundingSphere(geometry Geometry) (Geometry, error) {
```

Comment:
Do you not want ever to expand this function to return other geometry types? It's contained now in the kinematic base package so it fine, but it seems weird (long-term, not now) to input a config for a box in your base, or code `NewBox` in for go and get a sphere back.

Nothing to do here, just a question.

Reply:
I'm not sure I understand the question.  You mean write another function `BoundingPrism`? We could certainly do that, there just hasnt been a need to yet.

Reply:
The flow of adding a box geometry to your robot config and this function outputting a sphere no matter what you input is kind of weird/funny. But you're right. There is no need, currently, to give back other geometry types. But I would keep that in mind as you expose more motion/nav capabilities. 😄

Reply:
So this is only used in the differential drive kinematic base, which does not consider orientation in creating its plans.  The only safe thing to do imo is create a sphere to use for collision checking.  Technically the underlying base itself will always have the user configured geometry

And for PTGs we now use the user specified geometry